### PR TITLE
531 affichage d un fil d ariane au dessus de la liste des fichiers

### PR DESCRIPTION
**Pull request by sBouzols**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,5 @@
 .env.test.local
 .env.production.local
 
+# jest
+/coverage
```
Comment (ChamseddineBhd) [inline .gitignore]: add new Line

File: src/components/directory-tree-view.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable prettier/prettier */
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: why this line ?

File: src/components/directory-tree-view.js [hunk 3]
```diff
@@ -80,6 +89,8 @@ const DirectoryTreeView = ({ rootDirectory }) => {
     const intl = useIntl();
     const intlRef = useIntlRef();
 
+    ///////////////////////////////////
+    // Component initialization
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: /* Component initialization */ is a better  style i think...

File: src/components/directory-tree-view.js [hunk 4]
```diff
@@ -89,6 +100,7 @@ const DirectoryTreeView = ({ rootDirectory }) => {
         initialMapData[rootDirectory.elementUuid] = rootDirectoryCopy;
         setMapData(initialMapData);
     }, [rootDirectory]);
+    ///////////////////////////////////
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: remove unnecessary line (the end line of the hook can easily be detected)

File: src/components/directory-tree-view.js [hunk 5]
```diff
@@ -104,26 +116,62 @@ const DirectoryTreeView = ({ rootDirectory }) => {
         setOpenAddNewStudyDialog(true);
     };
 
+    ///////////////////////////////////
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: same for all comments

File: src/components/directory-tree-view.js [hunk 6]
```diff
@@ -104,26 +116,62 @@ const DirectoryTreeView = ({ rootDirectory }) => {
...
+
+    const updatePath = useCallback(
+        (nodeId) => {
+            let path = [];
+            buildPath(nodeId, path);
+            if (path != null) dispatch(setCurrentPath(path));
+        },
+        [buildPath, dispatch]
+    );
+    ///////////////////////////////////
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: ...

File: src/components/directory-tree-view.js [hunk 7]
```diff
@@ -104,26 +116,62 @@ const DirectoryTreeView = ({ rootDirectory }) => {
...
+
+    const updatecurrentChildren = useCallback(
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: updateCurrentChildren

File: src/components/directory-breadcrumbs.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        marginRight: theme.spacing(0.5),
+        display: 'flex',
+        alignItems: 'center',
+        textAlign: 'center',
+        width: '18px',
+        height: '18px',
+    },
+}));
+
+const DirectoryBreadcrumbs = ({ rootDirectory }) => {
```
Comment (sBouzols) [inline src/components/directory-breadcrumbs.js]: this props is useless, must remove it.

File: src/components/directory-breadcrumbs.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        ) {
+            return (
+                <Typography className={classes.directory} color="textPrimary">
+                    <FolderOpenIcon className={classes.icon} />
+                    {currentPath[currentPath.length - 1].elementName}
+                </Typography>
+            );
+        }
+    };
+    ///////////////////////////////////
```
Comment (ChamseddineBhd) [inline src/components/directory-breadcrumbs.js]: same

File: src/components/directory-breadcrumbs.js [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+const DirectoryBreadcrumbs = ({ rootDirectory }) => {
+    const classes = useStyles();
+    const dispatch = useDispatch();
+
+    const selectedDirectory = useSelector((state) => state.selectedDirectory);
+    const currentPath = useSelector((state) => state.currentPath);
+
+    ///////////////////////////////////
+    // Handle User interactions
```
Comment (ChamseddineBhd) [inline src/components/directory-breadcrumbs.js]: same

File: src/components/directory-tree-view.js [hunk 11]
```diff
@@ -190,6 +249,20 @@ const DirectoryTreeView = ({ rootDirectory }) => {
         [expandedRef]
     );
 
+    const updateToggle = useCallback(
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: toggleDirectory instead of updateToggle ?

File: src/components/directory-tree-view.js [hunk 12]
```diff
@@ -270,6 +327,43 @@ const DirectoryTreeView = ({ rootDirectory }) => {
...
+    );
+
+    ///////////////////////////////////
+    // Handle components synchronization
+    useEffect(() => {
+        // test if we handle this change in the good treeview by checking if breadCrumbDirectory is in this treeview
+        if (breadCrumbDirectory !== null && mapDataRef.current[breadCrumbDirectory] !== undefined) {
+            // // fetch content
+            handleSelect(breadCrumbDirectory, false);
+            dispatch(setBreadCrumbDirectory(null));
```
Comment (ChamseddineBhd) [inline src/components/directory-tree-view.js]: should we use selectedDirectory instead of breadCrumbDirectory ?

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

**Review by ChamseddineBhd — APPROVED** [review]

**Pull request merged**